src/lightning/pytorch/callbacks/lr_monitor.py

```diff
@@ -202,20 +212,22 @@ def _extract_stats(self, trainer: "pl.Trainer", interval: str) -> Dict[str, floa
 
         return latest_stat
 
-    def _get_lr_momentum_stat(self, optimizer: Optimizer, names: List[str]) -> Dict[str, float]:
-        lr_momentum_stat = {}
+    def _get_optimizer_stats(self, optimizer: Optimizer, names: List[str]) -> Dict[str, float]:
+        stats = {}
         param_groups = optimizer.param_groups
         use_betas = "betas" in optimizer.defaults
 
         for pg, name in zip(param_groups, names):
             lr = self._extract_lr(pg, name)
-            lr_momentum_stat.update(lr)
+            stats.update(lr)
             momentum = self._extract_momentum(
                 param_group=pg, name=name.replace(name, f"{name}-momentum"), use_betas=use_betas
             )
-            lr_momentum_stat.update(momentum)
+            stats.update(momentum)
+            weight_decay = self._extract_weight_decay(pg, name + "-weight_decay")
```

Comment:
nit: we use fstring directly above

```suggestion
            weight_decay = self._extract_weight_decay(pg, f"{name}-weight_decay")
```